src/test/java/teammates/logic/core/InstructorsLogicTest.java

```diff
@@ -646,4 +647,47 @@ private void testGetCoOwnersForCourse() {
                 && generatedCoOwnersEmails.containsAll(coOwnersEmailsFromDataBundle));
     }
 
+    private void testUpdateToEnsureValidityOfInstructorsForTheCourse() {
+        ______TS("Should not grant the currently being edited instructor the privilege of modifying instructors");
+
+        ______TS("The course has more than 1 instructor with modifying instructor privilege");
+        String courseId = "idOfTypicalCourse1";
+        InstructorAttributes instructorToUpdate =
+                InstructorAttributes.builder(courseId, "idOfInstructor4@gmail.com")
+                        .withGoogleId("idOfInstructor4")
+                        .withPrivileges(
+                                new InstructorPrivileges(
+                                        Const.InstructorPermissionRoleNames.INSTRUCTOR_PERMISSION_ROLE_TUTOR
+                                )
+                        ).build();
+        instructorsLogic.updateToEnsureValidityOfInstructorsForTheCourse(courseId, instructorToUpdate);
+
+        assertFalse(instructorToUpdate.privileges.isAllowedForPrivilege(Const.InstructorPermissions.CAN_MODIFY_INSTRUCTOR));
+
+        ______TS("The course has 1 registered instructor with modifying instructor privilege");
+        dataBundle.instructors.remove("instructorNotYetJoinCourse");
+        removeAndRestoreTypicalDataBundle();
```

Comment:
While this works, this is an overkill as what you do here is cleaning and recreating the DB just to remove one entity. You can simply use an available logic method (in this same class, in fact) to remove the said instructor.

Reply:
@wkurniawan07 I think using a method of `InstructorLogic` would create a dependency between the 2 methods which is not desirable for testing

Reply:
The same argument is multitudes times worse when applied to `removeAndRestoreTypicalDataBundle`.